### PR TITLE
ci: pin to ubuntu-22.04 for now

### PR DESCRIPTION
**Pull request by eitsupi**
See https://github.com/PRQL/prql/pull/4941#issuecomment-2402040831

So far, CI in this repository does not seem to support Ubuntu 24.04.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -519,6 +519,9 @@ jobs:
       - uses: baptiste0928/cargo-install@v3
         with:
           crate: cargo-msrv
+          # TODO: remove this version pinning
+          # The latest 0.16 supports workspace inheritance, so the check will fail
+          version: "0.15"
```
Comment (eitsupi) [inline .github/workflows/tests.yaml:524]: I think this is needed to pass the test for now.
Will be fixed by #4941

**Comment by eitsupi** [comment]
If we do not merge this, all CI for any PRs will fail, so I will merge for now.

**Comment by max-sixty** [comment]
Thanks a lot! I guess we need to add python onto the images.

(would be great if we could pin the images + have dependabot upgrade them, but seems that dependabot doesn't do this...)

**Comment by eitsupi** [comment]
> I guess we need to add python onto the images.

Python is included in ubuntu-24.04.
https://github.com/actions/runner-images/blob/ubuntu24/20240922.1/images/ubuntu/Ubuntu2404-Readme.md

Do you think we need to install something like libpython-dev with apt?

<details>
<summary>Details</summary>

```log
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/runner/.cargo-install/cargo-nextest/bin:/home/runner/.cargo-install/cargo-insta/bin:/snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin" VSLANG="1033" "cc" "-m64" "/tmp/rustctKOCmV/symbols.o" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/prqlc_python-9021ca62d3f7bd8b.prqlc_python.32c6953806f32434-cgu.0.rcgu.o" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/prqlc_python-9021ca62d3f7bd8b.prqlc_python.32c6953806f32434-cgu.1.rcgu.o" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/prqlc_python-9021ca62d3f7bd8b.prqlc_python.32c6953806f32434-cgu.2.rcgu.o" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/prqlc_python-9021ca62d3f7bd8b.aqn2ch64o6hjwlxd20i6q1in5.rcgu.o" "-Wl,--as-needed" "-L" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps" "-L" "/home/runner/work/prql/prql/target/debug/deps" "-L" "/usr/lib/x86_64-linux-gnu" "-L" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/build/libduckdb-sys-b06479e6b07a9cc5/out" "-L" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/build/zstd-sys-9cd23ce4dc9a27a9/out" "-L" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/build/libsqlite3-sys-4ad30827a17f4d7c/out" "-L" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/build/ring-56c60c7480b603da/out" "-L" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/build/psm-fb494b6a7cd63efb/out" "-L" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-95407a28b34cb7d8.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgetopts-6cf033bd0623f965.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunicode_width-92be448731e25aff.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_std-9c68449f4da60201.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libinsta-37b4bbf5b79fd0a7.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libwalkdir-e573ca162a6844aa.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libsame_file-72a178e4843c9577.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libglobset-6afdbd0bd3fc8acc.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libbstr-46893dbda5a3769c.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libconsole-0ec0581b9e16abb3.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/liblazy_static-bd9ff85acf5828b2.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libsimilar-965e5bf6c49efcc5.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libunicode_segmentation-6c0ac366ded6abb9.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/liblinked_hash_map-c54c37cd4d00f1d7.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3-68b2018844aa2dbc.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libanyhow-8c35de29bc840ba8.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libmemoffset-67cf73b33cc6ba3d.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-18481af656e2175b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libparking_lot_core-d77f144692018e67.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libsmallvec-8b0b22aa08883e33.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/liblock_api-7ee2e850b4937c40.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libscopeguard-c7ce9bd95f214829.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-84e0e1ece67e5e5d.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libunindent-697319a1a74acc6b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libprqlc-5c743d69303b650b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libsqlformat-3e0602b0c542e38b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libunicode_categories-fb00fc044db746ed.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libnom-1b59f71b52c81976.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libcsv-10d50c10fa1d106c.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libcsv_core-06181577458228da.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libsqlparser-d65418872b74ab7b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libariadne-97bd79ea74192566.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libunicode_width-25f921783cac09d6.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libyansi-343012e665d50596.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libchrono-740982b192e6692c.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libnum_traits-ee65781ddce2f2a7.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libiana_time_zone-3bc046359db93821.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libregex-075eca1e82b667c2.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libregex_automata-61ae3320612edc07.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libaho_corasick-8cae25cca24f9e68.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libregex_syntax-a70b0111690511fd.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libprqlc_parser-cdcb258ce254221b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libserde_yaml-338310c9e4eabdad.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libindexmap-50abedc04d833154.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libequivalent-2064f25a7c3e0fdf.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libhashbrown-4649576ed0860dc3.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libunsafe_libyaml-c861698f41b478e0.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/liblog-ae6404bb0cc9818a.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libitertools-4f5217b66cf8c755.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libeither-ff27cc1f192a7511.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libschemars-c5c848dc36cc90aa.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libdyn_clone-602f51bdc5ca8ed8.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libref_cast-364e5ea10d0655bf.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libserde_json-207a0cb1b12aa27b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libmemchr-7109c87bbfd6ff62.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libitoa-ea6aee116d398e7e.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libryu-294f07668d49512f.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libchumsky-9610f6a2b028f11c.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libstacker-033da7f98d268a9a.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libpsm-f16dc78b1de77fc6.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libhashbrown-ae704ffdab258e1f.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libahash-817a355d1145dbce.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libgetrandom-1fc0b24cfec4e402.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/liblibc-d4bbfba31b2b2b36.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libonce_cell-92186e6e342679a7.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libcfg_if-03e14143fe5887b5.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libzerocopy-2af2927c0872b9bf.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/liballocator_api2-d4b9ba1e3898f836.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libstrum-59e77847a2f51aea.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libsemver-b9283a291db57576.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libserde-1170b2415ef34459.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libanstream-84a9af77a48e8b1b.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libanstyle_query-f78ef7362b792fb3.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libis_terminal_polyfill-2307e434d6cb8f08.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libanstyle-0925051d1839ccf2.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libcolorchoice-5b26aa145bfed543.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libanstyle_parse-58b0891d216b97e3.rlib" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/libutf8parse-e78e7d1e7738cac0.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-52417a9a08ba8fb9.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-1b238e97929fdc18.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-3692c2663daa48a5.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-5c1e2d61611c0240.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-e55c5186b9bbabcb.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-7911d2fe92903c6c.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-c03510cdc48cfa52.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-7ec6f3765287c7b7.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-ab3f1b788cf79cfc.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-6db4226b4e34b1a6.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-f5bb1a38d410fa1e.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-e585efe74c64604d.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-03f8e93b59ffc316.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-669e9182bfe197a6.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-64b53226be979181.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-26921ac7e6a44d2b.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-6703049be165ebf1.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-04bfdf6b094564ce.rlib" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-6648dc218e522d87.rlib" "-Wl,-Bdynamic" "-lpython3.10" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/home/runner/.rustup/toolchains/1.80.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/runner/work/prql/prql/target/x86_64-unknown-linux-gnu/debug/deps/prqlc_python-9021ca62d3f7bd8b" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: /usr/bin/ld: cannot find -lpython3.10: No such file or directory
```

</details>

**Comment by max-sixty** [comment]
Ah, interesting. Possibly. I guess we'll hear from others fairly soon on what's required...

**Pull request merged**